client_subscribe.go

```diff
@@ -0,0 +1,50 @@
+package sdk
+
+import (
+	"context"
+	"fmt"
+
+	abci "github.com/tendermint/tendermint/abci/types"
+)
+
+var subscribeQuery = "tm.event='Tx' AND wasm._contract_address CONTAINS '%s' AND wasm.action='execute_claim'"
+
+func (c *Client) Subscribe(ctx context.Context, contractAddress string, acknowledge func(ctx context.Context, msg []abci.Event) error) error {
+	// Create context to control explicitly ws subscription
+	wsCtx, cancelWsCtx := context.WithCancel(context.Background())
```

Comment:
maybe just use the parent context. Or inherit the parent contex. Or why the local context is used?

Reply:
Because if parent context will done - I will loose ability to Unsubscribe from WS

I need: 

1) Close parent context, wait for acknowledge is done
2) Then Unsubscribe from WS

Reply:
unsubscribtion will be triggered  by `<-ctx.Done()`. Line 36. So what's the reason for additional context?

Reply:
Yep, but the problem is that if I will provide parent context here: 

```
	err = tendermintNode.Start(wsCtx)
	if err != nil {
		return fmt.Errorf("tendermintNode.Start(): %v", err)
	}
```
	
then I won't be able to call this:

```
			err = tendermintNode.UnsubscribeAll(wsCtx, "")
			if err != nil {
				return fmt.Errorf("tendermintNode.UnsubscribeAll: %v", err)
			}
```

Reply:
When context will be done, I will loose my ability to send UnsibscribeAll, because my ws connection will be closed)

Reply:
They adviced to unsubscribe in comment, so I think it's important thing:

```
	// The context only governs the initial subscription, it does not control
	// the lifetime of the channel. To cancel a subscription call Unsubscribe or
	// UnsubscribeAll.
```

Reply:
got it, thanks